compiler/src/main/java/com/squareup/anvil/compiler/codegen/reference/ClassReferenceIr.kt

```diff
@@ -0,0 +1,74 @@
+package com.squareup.anvil.compiler.codegen.reference
+
+import com.squareup.anvil.compiler.api.AnvilCompilationException
+import com.squareup.anvil.compiler.fqName
+import com.squareup.anvil.compiler.internal.reference.Visibility
+import com.squareup.anvil.compiler.internal.reference.Visibility.INTERNAL
+import com.squareup.anvil.compiler.internal.reference.Visibility.PRIVATE
+import com.squareup.anvil.compiler.internal.reference.Visibility.PROTECTED
+import com.squareup.anvil.compiler.internal.reference.Visibility.PUBLIC
+import org.jetbrains.kotlin.backend.common.extensions.IrPluginContext
+import org.jetbrains.kotlin.descriptors.DescriptorVisibilities
+import org.jetbrains.kotlin.ir.symbols.IrClassSymbol
+import org.jetbrains.kotlin.ir.util.isInterface
+import org.jetbrains.kotlin.name.FqName
+import kotlin.LazyThreadSafetyMode.NONE
+
+internal class ClassReferenceIr(
+  val clazz: IrClassSymbol,
+  val context: IrPluginContext
+) : AnnotatedReferenceIr {
+  val fqName: FqName by lazy { clazz.fqName }
```

Comment:
It's not necessary to make this lazy, because it's one of more commonly accessed properties. Especially not with double-checked locking.